### PR TITLE
NXP-25381: initialize WOPI from a discovery XML file

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIDiscovery.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    public NetZone getNetZone() {
+        return netZone;
+    }
+
+    public void setNetZone(NetZone netZone) {
+        this.netZone = netZone;
+    }
+
+    protected static XmlMapper XML_MAPPER = new XmlMapper();
```
Comment (kevinleturc) [inline nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIDiscovery.java]: `final` ?

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    // extension => wopi action => wopi action url
+    protected Map<String, Map<String, String>> extensionActionURLs = new HashMap<>();
+
+    @Override
+    public void start(ComponentContext context) {
+        File wopiDir = new File(Environment.getDefault().getData(), WOPI_DIR);
+        File discoveryFile = new File(wopiDir, DISCOVERY_XML);
+        if (!discoveryFile.exists() || !discoveryFile.isFile()) {
+            log.error("Cannot load discovery file at: {}, WOPI disabled.", discoveryFile.getAbsolutePath());
```
Comment (troger) [inline nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java]: Don't remember if it's a good idea to log at error level the absolute path (security wise)?

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    protected void loadDiscovery(File discoveryFile) {
+        WOPIDiscovery discovery = WOPIDiscovery.read(discoveryFile);
+        discovery.getNetZone().getApps().forEach(this::registerApp);
+    }
+
+    protected void registerApp(WOPIDiscovery.App app) {
+        app.getActions().forEach(action -> {
+            extensionAppNames.put(action.getExt(), app.getName());
+            extensionActionURLs.computeIfAbsent(action.getExt(), (k) -> new HashMap<>())
```
Comment (kevinleturc) [inline nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java]: We don't use parenthesis when single argument in Java.

File: nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java (hunk 2 above)
Comment (kevinleturc) [reply nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java]: I don't think so, furthermore as the location is static I don't see any blocker to just log something without the path.

**Comment by ataillefer** [comment]
:+1:

**Pull request merged**